VirtoCommerce.CartModule.Data/Repositories/CartRepositoryImpl.cs

```diff
@@ -247,6 +247,15 @@ public virtual void RemoveCarts(string[] ids)
             }
         }
 
+        public virtual void SoftRemoveCarts(string[] ids)
+        {
+            var carts = ShoppingCarts.Where(x => ids.Contains(x.Id));
```

Comment:
Check if `ids` is empty.

It would be perfect to avoid loading carts:
`update Cart set IsDeleted = 1 where Id in (...)`